python/across_api/base/fov.py

```diff
@@ -0,0 +1,382 @@
+        defined by the given parameters. This works by calculating a SkyCoord
+        containing every non-zero probability pixel, uses the
+        `in_fov` method to check which pixels are inside the FOV,
+        and then finding the integrated probability of those pixels.
+
+        Note: This method makes no attempt to deal with pixels that are only
+        partially inside the FOV, i.e. Earth occultation is calculated for
+        location of the center of each HEALPix pixel.
+
+        If `healpix_order` == "NUNIQ", it assumes that `healpix_loc` contains a
+        multi-order HEALPix map, and handles that accordingly.
+
+        Parameters
+        ----------
+        healpix_loc
+                An array containing the probability density values for each HEALPix
+                pixel.
+        healpix_nside
+                The NSIDE value of the HEALPix map. If not provided, it will be
+                calculated based on the length of healpix_loc.
+        healpix_order
+                The ordering scheme of the HEALPix map. Default is "NESTED".
+
+        Returns
+        -------
+        float
+                The amount of probability inside the FOV.
+
+        """
+        # Extract the NSIDE value from the HEALPix map, also level and ipix if
+        # this is a MOC map
+        if healpix_order == "NUNIQ":
+            level, ipix = ah.uniq_to_level_ipix(healpix_loc["UNIQ"])
+            uniq_nside = ah.level_to_nside(level)
+            healpix_loc = healpix_loc["PROBDENSITY"]
+        else:
+            nside = ah.npix_to_nside(len(healpix_loc))
+
+        # Find where in HEALPix map the probability is > 0
+        nonzero_prob_pixels = np.where(healpix_loc > 0.0)[0]
+
+        # Create a list of RA/Dec coordinates for these pixels
+        if healpix_order == "NUNIQ":
+            ra, dec = ah.healpix_to_lonlat(
+                ipix[nonzero_prob_pixels],
+                nside=uniq_nside[nonzero_prob_pixels],  # type: ignore
+                order="NESTED",
+            )
+        else:
+            ra, dec = ah.healpix_to_lonlat(
+                nonzero_prob_pixels, nside=nside, order=healpix_order
+            )
+
+        # Convert these coordinates into a SkyCoord
+        skycoord = SkyCoord(ra=ra, dec=dec, unit="deg")
+
+        # Calculate pixel indicies of the all the regions inside of the FOV
+        visible_probability_pixels = nonzero_prob_pixels[self.in_fov(skycoord=skycoord)]
+        # Calculate the amount of probability inside the FOV
+        if healpix_order == "NUNIQ":
+            # Calculate probability in FOV by multiplying the probability density by
+            # area of each pixel and summing up
+            pixarea = ah.nside_to_pixel_area(uniq_nside[visible_probability_pixels])
+            return float(
+                round(
+                    np.sum(healpix_loc[visible_probability_pixels] * pixarea.value), 5
+                )
+            )
```

Comment:
Why round it?

Reply:
Was in Jamie's original implementation. I changed the result to be rounded in the tests instead.